src/reasoner/ReasonerManager.cpp

```diff
@@ -53,8 +53,8 @@ std::vector<DefiningReasoner> ReasonerManager::findDefiningReasoner(const Predic
 		if (rdfClass) {
 			for (auto &x: goalDriven_) {
 				for (auto &definedClassIndicator : x.second->definedClasses()) {
-					auto definedClass = kb_->vocabulary()->defineClass(definedClassIndicator.functor()->stringForm());
-					if(definedClass->isSubClassOf(rdfClass)) {
```

Comment:
but the subclass of check is missing now!

Reply:
BTW: slightly more efficient: Only call `getDefinedClass` and check if it is a null pointer. Then you can avoid second call to vocabulary